### PR TITLE
[api-extractor] Fix an issue where "ae-undocumented" was incorrectly reported for private members

### PR DESCRIPTION
**Pull request by octogonz**
## Summary

Fix an edge case from PR https://github.com/microsoft/rushstack/pull/4370

 

## How it was tested

Manually created a test project that reproduces the bug reported in https://github.com/microsoft/rushstack/pull/4370#issuecomment-1760606108 then confirmed it is fixed.

Rebuilt the entire Rush Stack monorepo using my PR build of `api-extractor`, confirmed that no existing API reports have any changes.

### CODE REVIEW

**Review by D4N14L — APPROVED** [review]

File: apps/api-extractor/src/generators/ApiReportGenerator.ts [hunk 1]
```diff
@@ -427,6 +431,12 @@ export class ApiReportGenerator {
     }
   }
 
+  private static _shouldIncludeInReport(astDeclaration: AstDeclaration): boolean {
+    // Private declarations are not included in the API report
```
Comment (iclanton) [inline apps/api-extractor/src/generators/ApiReportGenerator.ts:435]: Move to a TSDoc comment for the function?

**Review by iclanton — APPROVED** [review]

**Comment by Josmithr** [comment]
@octogonz Looks like this PR is potentially ready to be merged. I'm hoping to consume these changes, so I figured I would ping 🙂

**Review by leordev — APPROVED** [review]
Tested the changes locally and it's working! Thank you so much again! 🚀

File: apps/api-extractor/src/generators/ApiReportGenerator.ts (hunk 1 above)
Comment (octogonz) [reply apps/api-extractor/src/generators/ApiReportGenerator.ts:435]: The comment describes the current implementation, not the behavior of the function.
